### PR TITLE
DEV-10143: Update SubawardSearch w/ County Cols

### PR DESCRIPTION
**Pull request by dpb-bah**
**Description:**
Adding several prime award county columns and updating the subaward county columns in `rpt.subaward_search` (having them be based on the zip data on the broker side instead of deriving it based on city and state data on USAS).

**Technical details:**
* Includes a migration to rpt.subaward_search

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
    - [ ] Frontend <OPTIONAL>
    - [ ] Operations <OPTIONAL>
    - [ ] Domain Expert <OPTIONAL>
4. [x] Matview impact assessment completed
5. [x] Frontend impact assessment completed
6. [x] Data validation completed
7. [x] Appropriate Operations ticket(s) created
8. [x] Jira Ticket [DEV-10143](https://federal-spending-transparency.atlassian.net/browse/DEV-10143):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected (API | Script | Download)
    - [x] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: usaspending_api/search/delta_models/subaward_search.py [hunk 1]
```diff
@@ -81,6 +87,8 @@
     "sub_place_of_perform_state_code": {"delta": "STRING", "postgres": "TEXT"},
     "sub_place_of_perform_state_name": {"delta": "STRING", "postgres": "TEXT"},
     "sub_place_of_performance_zip": {"delta": "STRING", "postgres": "TEXT"},
+    "sub_place_of_performance_county_code": {"delta": "STRING", "postgres": "TEXT"},
```
Comment (dporth-frb) [inline usaspending_api/search/delta_models/subaward_search.py:90]: Our code base hard codes the pattern of these these legal entity and performance location columns. It requires that they follow this pattern, "sub_place_of_perform_" and "sub_legal_entity_". This column name goes against the pattern and I anticipate it will cause issues. For example, see the file `/workspaces/usaspending-api/usaspending_api/search/v2/views/spending_by_geography.py` method `post` on line 70.

**Review by dporth-frb — CHANGES_REQUESTED** [review]

File: usaspending_api/search/delta_models/subaward_search.py (hunk 1 above)
Comment (dpb-bah) [reply usaspending_api/search/delta_models/subaward_search.py:90]: If that's the case, the downloads would need to update too right? Since they are already existing and pointing to the old names? Wondering why this would be a problem now and not already since they're already like this.

Comment (dpb-bah) [reply usaspending_api/search/delta_models/subaward_search.py:90]: Ah I see. The columns we were trying to update had that subtle name change. Fixing.

Comment (dpb-bah) [reply usaspending_api/search/delta_models/subaward_search.py:90]: Ah I see. The columns we were trying to update had that subtle name change. Good catch.

Comment (dpb-bah) [reply usaspending_api/search/delta_models/subaward_search.py]: With the new columns, I kept them as `place_of_performance_county_*` as we currently don't use them in the code but can change them to the `place_of_perform_county_*` versions. Do you have a preference?

Comment (dporth-frb) [reply usaspending_api/search/delta_models/subaward_search.py]: >  we currently don't use them in the code

That's true, but if we do end up using them in the places that hard code this pattern it'll matter.

I think `place_of_perform_` is good

File: usaspending_api/search/delta_models/subaward_search.py [hunk 2]
```diff
@@ -44,6 +46,8 @@
     "place_of_perform_state_code": {"delta": "STRING", "postgres": "TEXT"},
     "place_of_perform_state_name": {"delta": "STRING", "postgres": "TEXT"},
     "place_of_performance_zip": {"delta": "STRING", "postgres": "TEXT"},
+    "place_of_perform_county_code": {"delta": "STRING", "postgres": "TEXT"},
```
Comment (dporth-frb) [inline usaspending_api/search/delta_models/subaward_search.py:49]: Why did you move these key value pairs?

Comment (dpb-bah) [reply usaspending_api/search/delta_models/subaward_search.py:49]: Those columns were originally derived on the USASpending code and the order of the columns had those last. Now that they are pulled from the Broker, I moved them to the section where we're pulling other broker values directly (sans major transformations/derivations).

Comment (dpb-bah) [reply usaspending_api/search/delta_models/subaward_search.py:49]: I'm assuming you are referring to the sub_place_* and the sub_legal_*. The prime award county fields are new.

Comment (dporth-frb) [reply usaspending_api/search/delta_models/subaward_search.py:49]: Correct. Why does order matter?

Comment (dpb-bah) [reply usaspending_api/search/delta_models/subaward_search.py:49]: It groups the columns into their respective sections for clarity/simplicity (see the comments)

-- Broker Subaward Table Meta
-- Prime Award Fields (from Broker)
-- Subaward Fields (from Broker)
-- Additional Prime Award Fields (from Broker)
-- Additional Subaward Fields (from Broker)
-- USAS Links (and associated derivations)
-- USAS Derived Fields

So we're moving those updated sub county cols from USAS Derived Fields to Subaward Fields. The order of the models at the top must match the order of the columns selected in the populate query.

**Review by dporth-frb — APPROVED** [review]
